src/commands/dev.ts

```diff
@@ -353,9 +353,24 @@ export default class Dev extends BaseCommand {
       linked_components
     );
 
-    const port_available = await PortUtil.isPortAvailable(flags.port);
-    if (!port_available) {
-      this.error(`Could not run architect on port ${flags.port}.\nPlease stop an existing process or specify --port to choose a different port.`);
+    let port_available = await PortUtil.isPortAvailable(flags.port);
```

Comment:
Can we move this to its own function that just gets the available port?

This would make it easier to read and also would produce something like
```
const port = getAvailablePort(flags.port)
```
Now we just use port for the rest of the function and its immutable. Would also keeps our flags less mutable which is generally a good thing.

Reply:
Yes, makes sense.